File: LeetCode_Solutions/6-zigzag-conversion/zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        mat=['']*numRows
        k=0
        dir=(numRows==1)-1
        for i in s:
            mat[k]+=i
            if k==0 or k==numRows-1:
                dir*=-1
            k+=dir
        return ''.join(mat)

File: LeetCode_Solutions/6-zigzag-conversion/zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) {
        StringBuilder sb=new StringBuilder();
        List<Character>[] row=new List[numRows];
        int k=0;
        int dir=(numRows==1)?0:-1;
        for(int i=0;i<numRows;i++){
            row[i]=new ArrayList<>();
        }
        for(char c:s.toCharArray ()){
            row[k].add(c);
            if(k==0 || k==numRows-1){
                dir*=-1;
            }
            k+=dir;
        }
        for (List<Character> rows : row)
            for (final char c : rows)
                sb.append(c);
        return sb.toString();
    }
}